### ISSUE TITLE
Virtualbox 7.1 Support

### ISSUE THREAD

**Issue by speefak** [phpvirtualbox/phpvirtualbox#344]
Support this phpVB Version support VB 7.1 ?

**Comment by ediazrod** [comment]
My recomendation at this time is avoid phpvirtualbox to all cost..

if you need something remote go to remotebox.

As far I use phpvirtualbox you can find issues locks and many problems if you allow to phpvirtualbox configure your enviroment, Why? I don't know but after of find sooo many issues I stop usign this software.

Is a sad but...

**Comment by speefak** [comment]
The Cockpit looks quite good, but still lacks functions like VirtualBox. phpVirtualBox is currently only usable with version 7.0.14, where snapshots are supposedly created according to phpVirtualBox, but not actually by VirtualBox itself. At the moment, I use X-forwarding to directly run VirtualBox and use phpVirtualBox only for starting/cloning and for an overview. It's a pity, because VirtualBox in combination with phpVirtualBox used to be a top VM software. After over 10 years of using VirtualBox, I too am desperately looking for an alternative. Proxmox, ESXI, Freenas, etc., are too bloated, and Cockpit is not yet ready for productive use. Unfortunately, there aren't many alternatives for a server without X with web interfaces.

**Comment by studnitskiy** [comment]
I made a fork of a fork with 7.1 support - https://github.com/studnitskiy/phpvirtualbox
Core functionality should work fine, but I'm sure there are bugs I didn't get to

**Comment by Neustradamus** [comment]
@studnitskiy: Nice, good job!

**Comment by jofrey007** [comment]
Great job ! I hope you will continue with support for VB 7.1 :-)

**Comment by jofrey007** [comment]
> I made a fork of a fork with 7.1 support - https://github.com/studnitskiy/phpvirtualbox Core functionality should work fine, but I'm sure there are bugs I didn't get to


Great job ! I hope you will continue with support for VB 7.1 :-)

**Comment by httkl** [comment]
> I made a fork of a fork with 7.1 support - https://github.com/studnitskiy/phpvirtualbox
> Core functionality should work fine, but I'm sure there are bugs I didn't get to

Works flawlessly here in a multi-server environment. Big thank you for your efforts! :)

**Comment by antwal** [comment]
hi, the docker build not working

**Comment by studnitskiy** [comment]
@antwal That's not related to VB 7.1 support.
The Dockerfile is 10 years old, it's still set to use PHP 5.6 that's out of support for 6 years already. Try updating FROM in the Dockerfile to php:8.2-apache and see if it works

**Comment by speefak** [comment]
"First of all, thank you for the work on supporting VirtualBox 7.1. Everything works so far except for the snapshot section. Creating a snapshot runs into a blank screen, and there is no error message. Snapshots are not being created."

**Comment by antwal** [comment]
@studnitskiy I solved it in another way, downloading a ready-made docker for 7.0, it works perfectly even adding the sdk for 7.1. thanks.

**Comment by speefak** [comment]
Good morning, I prefer not to use Docker as it provides kernel-independent virtualization but is tied to the host kernel (version, architecture). Additionally, Docker creates too many configurations on the host, like numerous network interface types. That's why I chose VirtualBox. The solution to the snapshot issue is quite simple: https://github.com/phpvirtualbox/phpvirtualbox/issues/346#issuecomment-2601233294

**Comment by studnitskiy** [comment]
Hi @arkhipenko 
All I can think of based on that is a firewall rule.
Based on the php output I see it's Ubuntu, so try
`sudo systemctl stop ufw`
then see what happens

**Comment by arkhipenko** [comment]
@studnitskiy 
Alexander, thanks for getting back to me.
I tried to disable firewall, but it still did not work. 
What's interesting is that previous phpvirtualbox installation (that I used for years with version 6) connects without any problem. It crashes shortly after since the VirtualBox is 7, but at least it connects on the same machine. RemoteBox connects as well. 
Any idea where I should be looking to? RemoteBox is a workaround, but I really like phpvirtualbox... 
I know PHP a little - if you point me in the direction of where this error might be coming from, I might be able to dig a little deeper.
Thank you in advance.

**Comment by arkhipenko** [comment]
OK. Seems like I found the issue - 
my connect string was `http://localhost:18083`, and that did not work. 
changing it to the IP of the NIC worked. Go figure.
I should be listening on all ips... and yet. 

```
tcp        0      0 omicron:18083           0.0.0.0:*               LISTEN
```

Thank you for your help!

**Comment by studnitskiy** [comment]
@arkhipenko 
Right, I overlooked that the local address in the netstat output is "omicron:18083". I saw "0.0.0.0", but didn't realize that's the remote address column without the headers.

vboxwebsrv will bind to the first address the -H parameter resolves into, in your case it's apparently the NIC's address.
Try configuring vboxwebsrv -H parameter with 0.0.0.0 IP instead of the hostname if you want it to respond on any local IPv4 address.

Also, what I found is that "-H localhost" might be misleading as well.
The connection attempt in the error message indicates "127.0.0.1", not "localhost".
Historically, localhost indeed resolved into 127.0.0.1. However, with IPv6 enabled (and it is on most systems these days) I found that "localhost" resolves into IPv6 address ::1 first, and so that's where vboxwebsrv binds when you specify -H localhost. Connections by IP 127.0.0.1:18083 won't work because the listening address is ::1:18083. Connecting by "localhost:18083" should work, assuming the client supports IPv6.

**Comment by arkhipenko** [comment]
Thanks again!
I am fine with it running on a NIC address. As long as it runs!

**Comment by trasherdk** [comment]
Check your `/etc/hosts` file. The order of `localhost` entries are important.

```
# For loopbacking.
127.0.0.1               localhost
::1                     localhost
```

**Comment by anonie-muss** [comment]
The changes from [studnitskiy](https://github.com/studnitskiy)'s fork (thank you!!!!!!!!!!!) have been incorporated and 7.1 is now supported.

**Comment by Neustradamus** [comment]
@anonie-muss: You are in the team?

Now, we can progress in the future.

Can you add you in public here:
- https://github.com/orgs/phpvirtualbox/people

Can you manage the Sourceforge too?
- https://sourceforge.net/projects/phpvirtualbox/

Note: Badly, the project has not a perfect history, that I have planned several months ago.
There were a lot of changes since latest phpVirtualBox official version: a lot of VirtualBox missing versions.

Thanks in advance.

Linked to:
- https://github.com/phpvirtualbox/phpvirtualbox/pull/352

**Comment by Neustradamus** [comment]
phpVirtualBox 7.1-1 Announcement done here:
- https://mastodon.social/@neustradamus/114423508825888266
- https://bsky.app/profile/neustradamus.bsky.social/post/3lnybvgu26c2m
- https://x.com/neustradamus/status/1917338853909881194

**Comment by Neustradamus** [comment]
phpVirtualBox 7.1-1 Announcement has been done here:
- https://mastodon.social/@neustradamus/114423508825888266
- https://bsky.app/profile/neustradamus.bsky.social/post/3lnybvgu26c2m
- https://x.com/neustradamus/status/1917338853909881194
- https://news.ycombinator.com/item?id=43838587
- https://www.reddit.com/r/virtualbox/comments/1kb0gan/phpvirtualbox_711_released/
